File: 515_find_largest_value_in_each_tree_row.java
class Solution {
    public List<Integer> largestValues(TreeNode r) {
        List<Integer> a = new ArrayList<>();
        if (r == null) return a;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(r);
        while (!q.isEmpty()) {
            int b = q.size(), c = Integer.MIN_VALUE;
            while (b-- > 0) {
                TreeNode d = q.poll();
                c = Math.max(c, d.val);
                if (d.left != null) q.add(d.left);
                if (d.right != null) q.add(d.right);
            }
            a.add(c);
        }
        return a;
    }
}

File: 515_find_largest_value_in_each_tree_row.cpp
class Solution {
public:
    vector<int> largestValues(TreeNode* z) {
        vector<int> a;
        if (!z) return a;
        queue<TreeNode*> b;
        b.push(z);
        while (!b.empty()) {
            int c = b.size(), d = INT_MIN;
            while (c--) {
                auto e = b.front(); b.pop();
                d = max(d, e->val);
                if (e->left) b.push(e->left);
                if (e->right) b.push(e->right);
            }
            a.push_back(d);
        }
        return a;
    }
};
